source/app.d

```diff
@@ -70,11 +70,48 @@ void defaultInit(UserManController userMan, DubRegistry registry)
 	}
 }
 
+struct AppConfig
+{
+	string ghuser, ghpassword,
+		   glurl, glauth,
+		   bbuser, bbpassword;
+	void init()
+	{
+		import dub.internal.utils : jsonFromFile;
+		auto regsettingsjson = jsonFromFile(NativePath("settings.json"), true);
+		static immutable variables = [
+			["ghuser", "github-user"],
+			["ghpassword", "github-password"],
+			["glurl", "gitlab-url"],
+			["glauth", "gitlab-auth"],
+			["bbuser", "bitbucket-user"],
+			["bbpassword", "bitbucket-password"],
+		];
+		static foreach (var; variables)
+		{
+			// fallback to environment variables
+			mixin(var[0] ~ ` = regsettingsjson["`~var[1]~`"].opt!string(
+				environment.get("`~var[1]~`".replace("-", "_").toUpper)
+			);`);
+		}
+	}
+}
+
 void main()
 {
 	bool noMonitoring;
 	setLogFile("log.txt", LogLevel.diagnostic);
 
+    version (linux)
+    {
+    	// register memory error handler on heroku
+    	if ("DYNO" in environment)
+    	{
+			import etc.linux.memoryerror : registerMemoryErrorHandler;
+        	registerMemoryErrorHandler();
```

Comment:
This was very helpful during debugging. It might even make sense to always activate this, but for now I only enabled this for Heroku (there's no way to attach `gdb` to your instance or get a coredump anyhow).